tensorflow_serving/batching/batching_session.cc

```diff
@@ -79,6 +81,34 @@ TensorSignature TensorSignatureFromRunArgs(
   return signature;
 }
 
+// Constructs vector of all task inputs from Batch of BatchingSessionTasks.
+// Input for each task is a vector of pairs (tensor_name, tensor_value).
+std::vector<std::vector<std::pair<string, Tensor>>> GetTaskInputsVector(
+    const Batch<BatchingSessionTask>& batch) {
+  std::vector<std::vector<std::pair<string, Tensor>>> all_task_inputs;
+  for (int i = 0; i < batch.num_tasks(); ++i) {
+    const std::vector<std::pair<string, Tensor>>& task_inputs =
+      *batch.task(i).inputs;
+    all_task_inputs.push_back(task_inputs);
+  }
+  return all_task_inputs;
+}
+// Returns true iff all dims of shape1 are equal to dims of shape2 starting with
+// the first dimension.
```

Comment:
Clarify that you are counting dimensions starting from zero, so maybe "starting with the first (not zeroth) dimension"